python/extension/CUDAQuantumExtension.cpp

```diff
@@ -178,6 +178,42 @@ PYBIND11_MODULE(_quakeDialects, m) {
       "ORCA's backends",
       py::arg("input_state"), py::arg("loop_lengths"), py::arg("bs_angles"),
       py::arg("n_samples") = 10000);
+
+  auto photonicsSubmodule = cudaqRuntime.def_submodule("photonics");
+  photonicsSubmodule.def(
+      "allocate_qudit",
+      [](std::size_t &level) {
+        return cudaq::getExecutionManager()->allocateQudit(level);
+      },
+      "Allocate a qudit of given level.", py::arg("level"));
+  photonicsSubmodule.def(
+      "apply_operation",
+      [](const std::string &name, std::vector<double> &params,
+         std::vector<std::vector<std::size_t>> &targets) {
+        std::vector<cudaq::QuditInfo> targetInfo;
+        for (auto t : targets) {
+          targetInfo.push_back(cudaq::QuditInfo(t[0], t[1]));
+        }
```

Comment:
This would produce fewer intermediate copies.
```suggestion
        for (auto &t : targets)
          targetInfo.emplace_back(t[0], t[1]);
```